src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -248,8 +248,22 @@ private function validateAndDenormalize($currentClass, $attribute, $data, $forma
                 return;
             }
 
-            $builtinType = $type->getBuiltinType();
-            $class = $type->getClassName();
+            if (
+                $type->isCollection() &&
+                null !== ($collectionValueType = $type->getCollectionValueType()) &&
+                Type::BUILTIN_TYPE_OBJECT === ($collectionValueBuiltinType = $collectionValueType->getBuiltinType())
+            ) {
+                $builtinType = Type::BUILTIN_TYPE_OBJECT;
+                $class = $collectionValueType->getClassName().'[]';
+
+                if (null !== ($collectionKeyType = $type->getCollectionKeyType())) {
```

Comment:
if it's a standard in Symfony code base then http://symfony.com/doc/current/contributing/code/standards.html should be updated accordingly as well, the code sample L78 implies otherwise